File: hackerrank/easy/0054-beautiful-triplets/python/solution.py
## solution.py

from typing import List

def beautifulTriplets(d: int, arr: List[int]) -> int:
    values = set(arr)
    count = 0

    for x in arr:
        if x + d in values and x + 2 * d in values:
            count += 1

    return count


# --- demo run ---
if __name__ == "__main__":
    d_sample = 3
    arr_sample = [1, 2, 4, 5, 7, 8, 10]  # Expected: 3
    print(beautifulTriplets(d_sample, arr_sample))

File: hackerrank/easy/0054-beautiful-triplets/js/solution.mjs
// solution.mjs

export function beautifulTriplets(d, arr) {
  const values = new Set(arr);
  let count = 0;

  for (const x of arr) {
    if (values.has(x + d) && values.has(x + 2 * d)) {
      count++;
    }
  }

  return count;
}

// --- demo run ---
if (import.meta.url === `file://${process.argv[1]}`) {
  const dSample = 3;
  const arrSample = [1, 2, 4, 5, 7, 8, 10]; // Expected: 3
  console.log(beautifulTriplets(dSample, arrSample));
}
